conda_forge_admin_requests/cfep3_copy.py

```diff
@@ -7,50 +7,65 @@
 import subprocess
 from typing import Dict, Any
 
-
 import requests
 
+from .utils import split_label_from_channel, parse_filename
+
+
+def check_one(package: str, sha256: str):
+    if not isinstance(sha256, str) or len(sha256) != 64:
+        raise ValueError(
+            f"Key '{sha256}' must be SHA256 for the artifact (64 hexadecimal characters)"
+        )
+    
+    channel, subdir, artifact = package.rsplit("/", 3)
+    channel, _ = split_label_from_channel(channel)
+    pkg_name, version, _, _ = parse_filename(artifact)
+
+    # Check existence
+    url = f"https://conda-web.anaconda.org/{channel}/{subdir}/{artifact}"
+    r = requests.head(url)
+    if not r.ok:
+        raise ValueError(f"Package '{package}' at {channel} does not seem to exist")
+    
+    # Check SHA256
+    r = requests.get(
+        f"https://api.anaconda.org/dist/{channel}/{pkg_name}/{version}/{subdir}/{artifact}",
+        timeout=10,
+    )
+    r.raise_for_status()
+    api_sha256 = r.json()["sha256"]
+    if sha256 != api_sha256:
```

Comment:
```suggestion
    if not hmac.compare_digest(sha256, api_sha256):
```

You should always use this function when comparing digests to prevent timing attacks: https://docs.python.org/3/library/hmac.html#hmac.compare_digest